registry_resolver/src/registry_client.rs

```diff
@@ -1,5 +1,5 @@
 pub mod registry {
-    include!(concat!(env!("OUT_DIR"), "/registry_api.v1.rs"));
+    include!(concat!(env!("OUT_DIR"), "/gen/pb/registry_api.v1.rs"));
```

Comment:
Correct me if I'm wrong but [`OUT_DIR` implies generated code](https://doc.rust-lang.org/cargo/reference/environment-variables.html#environment-variables-cargo-sets-for-crates).
```suggestion
    include!(concat!(env!("OUT_DIR"), "/pb/registry_api.v1.rs"));
```

Reply:
The documentation says:
```
OUT_DIR — the folder in which all output and intermediate artifacts should be placed. This folder is inside the build directory for the package being built, and it is unique for the package in question.
```
which implies to me that there is a distinction between "output" and "intermediate" artifacts. So I think there's value in having a separate `gen` directory that is specifically for output artifacts that need to get imported into code. In fact `protofetch` puts its protobuf directory structure, which we can think of as an intermediate artifact, into `OUT_DIR` but not under `gen`, so someone looking at this code later can assume that `gen` only has rust sourcefiles and the other stuff in the `OUT_DIR` directory is not immediately relevant to them if they just care about the generated rust. So I think keeping the `gen` makes sense.

Reply:
makes perfect sense